components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SPInitSSOAuthnRequestValidator.java

```diff
@@ -59,49 +54,49 @@ public SAMLSSOReqValidationResponseDTO validate() throws IdentityException {
 
-            // Validate the version
-            if (!(authnReq.getVersion().equals(SAMLVersion.VERSION_20))) {
-                String errorResp =
-                        buildErrorResponse(SAMLSSOConstants.StatusCodes.VERSION_MISMATCH,
-                                "Invalid SAML Version in Authentication Request. SAML Version should be equal to 2.0");
-                if (log.isDebugEnabled()) {
-                    log.debug("Invalid version in the SAMLRequest" + authnReq.getVersion());
-                }
-                validationResponse.setResponse(errorResp);
-                validationResponse.setValid(false);
-                return validationResponse;
-            }
-
-            // Issuer MUST NOT be null
-            if (issuer.getValue() != null) {
-                validationResponse.setIssuer(issuer.getValue());
-            } else if (issuer.getSPProvidedID() != null) {
-                validationResponse.setIssuer(issuer.getSPProvidedID());
-            } else {
-                validationResponse.setValid(false);
-                String errorResp =
-                        buildErrorResponse(SAMLSSOConstants.StatusCodes.REQUESTOR_ERROR,
-                                "Issuer/ProviderName should not be empty in the Authentication Request.");
-                log.debug("SAML Request issuer validation failed. Issuer should not be empty");
-                validationResponse.setResponse(errorResp);
-                validationResponse.setValid(false);
-                return validationResponse;
-            }
-
-            // Issuer Format attribute
-            if ((issuer.getFormat() != null) &&
-                    !(issuer.getFormat().equals(SAMLSSOConstants.Attribute.ISSUER_FORMAT))) {
-                validationResponse.setValid(false);
-                String errorResp =
-                        buildErrorResponse(SAMLSSOConstants.StatusCodes.REQUESTOR_ERROR,
-                                "Issuer Format attribute value is invalid");
-                if (log.isDebugEnabled()) {
-                    log.debug("Invalid Issuer Format attribute value " + issuer.getFormat());
-                }
-                validationResponse.setResponse(errorResp);
-                validationResponse.setValid(false);
-                return validationResponse;
-            }
+			// Validate the version
+			if (!(authnReq.getVersion().equals(SAMLVersion.VERSION_20))) {
+				String errorResp = SAMLSSOUtil.buildErrorResponse(
+                        SAMLSSOConstants.StatusCodes.VERSION_MISMATCH,
+				        "Invalid SAML Version in Authentication Request. SAML Version should be equal to 2.0");
+				if (log.isDebugEnabled()) {
+					log.debug("Invalid version in the SAMLRequest" + authnReq.getVersion());
+				}
+				validationResponse.setResponse(errorResp);
+				validationResponse.setValid(false);
+				return validationResponse;
+			}
+
+			// Issuer MUST NOT be null
+			if (issuer.getValue() != null) {
```

Comment:
Use    "if (!SAMLVersion.VERSION_20.equals((authnReq.getVersion()))) {" to avoid null pointer exeptions